src/backend/executor/execSRF.c

```diff
@@ -972,3 +978,38 @@ tupledesc_match(TupleDesc dst_tupdesc, TupleDesc src_tupdesc)
 							   i + 1)));
 	}
 }
+
+
+void
+ExecSquelchFunctionResultSet(SetExprState *fcache,
+							 ExprContext *econtext,
+							 MemoryContext *argcontext)
+{
+	FunctionCallInfo fcinfo = fcache->fcinfo;
+
+	/*
+	 * if fcache->shutdown_reg  is set to true, the function is known to be
+	 * called at least once. Otherwise it was not called and it has no need to
+	 * release resources.
+	 */
+	if (SRF_IS_FIRSTCALL() || !fcache->shutdown_reg || !fcache->isSquelchSupported)
+	{
+		return;
+	}
+
+	/* Prepare a resultinfo node for communication. */
+	ReturnSetInfo rsinfo = {
```

Comment:
```suggestion
	ReturnSetInfo rsinfo =
	{
```

Reply:
or even use InitFunctionCallInfoData

Reply:
pgindent does not agree. Any reasons for that changes? 

> or even use InitFunctionCallInfoData
It's used only from one place, why introduce new entity?  «Entia non sunt multiplicanda praeter necessitatem»

Reply:
What is this new entity? I'm just suggesting shortening and beautifying the code.

Reply:
Right, macro InitFunctionCallInfoData does exists. But I can not agree that using macro with unnamed argument list makes code beautiful and clean. This code was initially copied from the same file https://github.com/arenadata/gpdb/blob/8a8177e6aa27c46a4cc7d85a06d818fd7422101d/src/backend/executor/execSRF.c#L147     https://github.com/arenadata/gpdb/blob/8a8177e6aa27c46a4cc7d85a06d818fd7422101d/src/backend/executor/execSRF.c#L588  but now we have C99 compiler so it's possible to make it even more descriptive.

Reply:
The contents of the macro can change later and then we will not have to edit unnecessary places.

Reply:
In that case we, or author of the change will have to edit place I mentioned above. Don't you suggest to change them as well?